### PR TITLE
chore: Update directory paths in README files

### PR DESCRIPTION
**Pull request by jasinliu**

### CODE REVIEW

**Review by acezen — APPROVED** [review]
Thanks for the fix.

File: cpp/README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ repository and navigated to the ``cpp`` subdirectory:
 
 ```bash
     $ git clone https://github.com/apache/incubator-graphar.git
-    $ cd GraphAr
+    $ cd incubator-graphar
```
Comment (liuxiaocs7) [inline cpp/README.md:45]: how about

```
$ git clone https://github.com/apache/graphar.git
$ cd graphar
```

Comment (acezen) [reply cpp/README.md:45]: incubator-graphar is ok

**Comment by acezen** [comment]
The CI failed is because `sudo apt-get update` got failed in github host, see https://github.com/orgs/community/discussions/120966

it's not related to this PR.

**Pull request merged**